packages/bitcore-node/src/modules/bitcoin/VerificationPeer.ts

```diff
@@ -71,10 +72,22 @@ export class VerificationPeer extends BitcoinP2PWorker {
     });
   }
 
-  async resync(from: number, to: number) {
+  async getBlockForNumber(currentHeight: number) {
     const { chain, network } = this;
-    let currentHeight = Math.max(1, from);
-    while (currentHeight < to) {
+    const locatorHashes = await ChainStateProvider.getLocatorHashes({
+      chain,
+      network,
+      startHeight: Math.max(1, currentHeight - 30),
```

Comment:
where this 30 is coming from? Why we should return a hash that is not the provided Height?

Reply:
This one is really tricky actually, it comes from https://github.com/bitpay/bitcore/blob/22d121cf9dbe180fc7e4599acf35bd90f9164994/packages/bitcore-node/src/providers/chain-state/internal/internal.ts#L553

Basically getLocatorHashes returns 30 hashes, and the most recent one is first in the list. 

So to use that to get the current hash, you'd need to start from 30 blocks in the past.

Reply:
tricky, indeed :)